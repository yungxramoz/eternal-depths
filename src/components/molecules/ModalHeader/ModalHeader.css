.modal-header {
  width: stretch;
  display: flex;
  flex-direction: column;
}

.dismiss-button {
  align-self: flex-end;
  margin: 0.5em;
}
