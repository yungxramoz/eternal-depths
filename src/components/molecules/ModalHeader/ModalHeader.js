import React from 'react'
import RpgSeparator from '../../templates/RpgSeperator/RpgSeparator'
import IconButton from '../IconButton/IconButton'
import './ModalHeader.css'

const ModalHeader = ({ title, dismissible, onDismiss }) => (
  <div className="modal-header">
    {dismissible && (
      <IconButton
        className="dismiss-button"
        size="large"
        icon="close"
        onClick={onDismiss}
      />
    )}
    <h2>{title}</h2>
    <RpgSeparator golden={false} />
  </div>
)

export default ModalHeader
