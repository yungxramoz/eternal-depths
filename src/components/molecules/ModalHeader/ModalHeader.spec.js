import { render } from '@testing-library/react'
import React from 'react'
import ModalHeader from './ModalHeader'

describe('ModalHeader', () => {
  it('renders correctly', () => {
    render(<ModalHeader />)
    expect(document.body).toMatchSnapshot('ModalBody')
  })
})
