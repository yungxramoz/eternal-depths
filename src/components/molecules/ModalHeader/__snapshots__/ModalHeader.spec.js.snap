// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader renders correctly: ModalBody 1`] = `
<body>
  <div>
    <div
      class="modal-header"
    >
      <h2 />
      <hr
        class="separator"
      />
    </div>
  </div>
</body>
`;
